app/utils/transformers/IncidentTransformer.scala

```diff
@@ -17,30 +17,40 @@
 package utils.transformers
 
 import connectors.ReferenceDataConnector
-import models.reference.Incident
+import generated.EndorsementType03
+import models.reference.{Country, Incident}
 import models.{Index, UserAnswers}
-import pages.incident.{IncidentCodePage, IncidentTextPage}
+import pages.incident.{EndorsementCountryPage, IncidentCodePage, IncidentTextPage}
 import uk.gov.hmrc.http.HeaderCarrier
 
 import javax.inject.Inject
 import scala.concurrent.{ExecutionContext, Future}
 
-class IncidentTransformer @Inject() (referenceDataConnector: ReferenceDataConnector)(implicit ec: ExecutionContext) extends PageTransformer {
+class IncidentTransformer @Inject() (referenceDataConnector: ReferenceDataConnector)(implicit
+  ec: ExecutionContext
+) extends PageTransformer {
 
   private case class TempIncident(
     typeValue: Incident,
-    text: String
+    text: String,
+    endorsement: Option[EndorsementType03],
+    endorsementCountry: Option[Country]
   )
 
   def transform(incidents: Seq[generated.IncidentType04])(implicit hc: HeaderCarrier): UserAnswers => Future[UserAnswers] = userAnswers => {
 
     lazy val incidentRefLookups = incidents.map {
       incidentType0 =>
-        referenceDataConnector
-          .getIncidentType(incidentType0.code)
-          .map(
-            incident => TempIncident(incident, incidentType0.text)
-          )
+        for {
+          incident <- referenceDataConnector
+            .getIncidentType(incidentType0.code)
+          country <- incidentType0.Endorsement
+            .map(
+              endorsementType0 => referenceDataConnector.getCountry(endorsementType0.country)
+            )
+            .map(_.map(Some(_)))
+            .getOrElse(Future.successful(None))
+        } yield TempIncident(incident, incidentType0.text, incidentType0.Endorsement, country)
```

Comment:
Defining the futures first before flatMapping over them in the for-comprehension allows them to run in parallel
https://stackoverflow.com/questions/19045936/scalas-for-comprehension-with-futures

Reply:
Also very minor but I wonder if
```
        for {
          incident <- referenceDataConnector.getIncidentType(incidentType0.code)
          country <- incidentType0.Endorsement match {
            case Some(value) => referenceDataConnector.getCountry(value.country).map(Some(_))
            case None => Future.successful(None)
          }
        }
```
looks a bit nicer?

Reply:
And combining that with the first comment:
```
        val incidentF = referenceDataConnector.getIncidentType(incidentType0.code)
        val countryF = incidentType0.Endorsement.map(_.country).lookup(referenceDataConnector.getCountry)
        for {
          incident <- incidentF
          country <- countryF
        } yield ...
```